### PR TITLE
CMakeLists: trim rbd/rgw forced dependencies

### PR DESCRIPTION
**Pull request by batrick**
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

File: src/osdc/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,9 @@
 set(osdc_files
-  Objecter.cc
-  Filer.cc)
-set(osdc_rbd_files
+  Filer.cc
```
Comment (dillaman) [inline src/osdc/CMakeLists.txt]: Aren't these already linked into librados? This could have the possibility of causing oddities.

Comment (batrick) [inline src/osdc/CMakeLists.txt]: I may be misunderstanding the intent of this code but I believe there was no reason to break these `osdc_rbd_files` into a separate group which is why I merged them. Nothing else was referring to `osdc_rbd_files`.

Comment (dillaman) [inline src/osdc/CMakeLists.txt]: OK -- if nothing else was referring to them, sounds good to me.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-07-27_04:52:14-rados-wip-kefu-testing-distro-basic-mira/

**Pull request merged**